cpp/include/raft/spectral/lanczos.hpp

```diff
@@ -0,0 +1,1228 @@
+/*
+ * Copyright (c) 2020, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+//for cmath:
+#define _USE_MATH_DEFINES
+
+#include <cmath>
+#include <vector>
+
+#include <cuda.h>
+#include <curand.h>
+
+#include <raft/cudart_utils.h>
+#include <raft/linalg/cublas_wrappers.h>
+#include <raft/handle.hpp>
+#include <raft/spectral/lapack.hpp>
+#include <raft/spectral/matrix_wrappers.hpp>
+#include <raft/spectral/warn_dbg.hpp>
+
+namespace raft {
+
+using namespace matrix;
+using namespace linalg;
+
+namespace spectral {
+
+// curandGeneratorNormalX
+inline curandStatus_t curandGenerateNormalX(curandGenerator_t generator,
```

Comment:
Just a note: we (cuml) will soon be bringing over a nice `rng` API built on top curand that should remove the need to invoke curand directly.